http/client.go

```diff
@@ -0,0 +1,186 @@
+package http
```

Comment:
Thanks for this file, I think it may be useful for some client library work I plan to do in the future. Factoring out this code may be useful to me. However, I prefer the code in the `http/` directory to be just for `rqlited`. So can you move this to the CLI directory instead? I'd the following path: `cmd/rqlite/http/client`.

If this client file turns out to work well for the CLI, I would actually make it a core part of a new client library -- which the CLI would import. Make sense?

Reply:
Sure sounds reasonable. 

I am still not that familiar with the code base, so I put the client in the first directory that seemed logical, I didn't know the code for the cli is all grouped under `cmd/rqlite` :sweat_smile: